source/PlayerInfo.cpp

```diff
@@ -1205,18 +1205,32 @@ const Ship *PlayerInfo::GiftShip(const Ship *model, const string &name, const st
 
 
 // Sell the given ship (if it belongs to the player).
-void PlayerInfo::SellShip(const Ship *selected)
+void PlayerInfo::SellShip(const Ship *selected, bool toStorage)
 {
 	for(auto it = ships.begin(); it != ships.end(); ++it)
 		if(it->get() == selected)
 		{
 			int day = date.DaysSinceEpoch();
-			int64_t cost = depreciation.Value(*selected, day);
+			int64_t cost;
+
+			if(toStorage)
+				cost = depreciation.Value(selected, day);
+			else
+				cost = depreciation.Value(*selected, day);
```

Comment:
I assume this calls to a different function in both cases (right?) but a comment couldnt hurt?
it being a trinary operation or not is up to personal preference
(I was thinking:)
cost = depreciation.Value(toStorage ? selected : *selected, day);

Reply:
I believe that a ternary operator needs to return the same type from either outcome.

Reply:
oooh true you'd have to make it ugly then with the whole line repeated
still could use a comment ^^

Reply:
Added a comment describing how passing a pointer or a reference changes the return value.